src/app/modules/persistence-service/persistence.service.spec.ts

```diff
@@ -38,7 +40,184 @@ describe('PersistenceService', () => {
+      // given
+      const tclFilePath = 'path/to/file?.tcl';
+      const tclFileTarget = 'newpath/to/file?.tcl';
+
+      // when
+      persistenceService.copyResource(tclFileTarget, tclFilePath)
+
+      // then
+      .then((response) => expect(response).toBe(tclFileTarget));
+      tick();
+
+      httpMock.match(pullMatcher)[0].flush({ failure: false, diffExists: false, headCommit: 'abcdef',
+        changedResources: [],
+        backedUpResources: []
+      });
+      tick();
+
+      httpMock.match(copyMatcher)[0].flush(tclFileTarget);
+  })));
+
+  it('expect messages about changes when copying and pull reports changes', fakeAsync(inject([HttpTestingController, PersistenceService],
+    (httpMock: HttpTestingController, persistenceService: PersistenceService) => {
+
+      // given
+      const tclFilePath = 'path/to/file?.tcl';
+      const tclFileTarget = 'newpath/to/file?.tcl';
+      let changedFiles: string[];
+      let backedUpFiles: BackupEntry[];
+
+      // when
+      messagingService.subscribe('files.changed', (files) => { changedFiles = files; });
+      messagingService.subscribe('files.backedup', (files) => { backedUpFiles = files; });
+      persistenceService.copyResource(tclFileTarget, tclFilePath)
+
+       // then
+      .then((response) => expect(response).toBe(tclFileTarget));
+      tick();
+
+      httpMock.match(pullMatcher)[0].flush({ failure: false, diffExists: true, headCommit: 'abcdef',
+        changedResources: [ 'some/file' ],
+        backedUpResources: [ { resource: 'some/other/file', backupResource: 'some/other/backup.file' } ]
+      });
+      tick();
+
+      httpMock.match(copyMatcher)[0].flush(tclFileTarget);
+      tick();
+
+      // messages where received and changedFiles and backedUpFiles were set
+      expect(changedFiles).toContain('some/file');
+      expect(backedUpFiles.length).toEqual(1);
+      expect(backedUpFiles[0].resource).toEqual('some/other/file');
+      expect(backedUpFiles[0].backupResource).toEqual('some/other/backup.file');
+   })));
+
+  it('ensure that pull passes no more "resources" and "dirtyResources" if closing/saved/non-dirty messages were received',
+      fakeAsync(inject([HttpTestingController, PersistenceService],
+                       (httpMock: HttpTestingController, persistenceService: PersistenceService) => {
+      // given
+      persistenceService.startSubscriptions();
+
+      // when
+      messagingService.publish('navigation.open', 'some/file');
+      messagingService.publish('navigation.open', 'some/other/file');
+      messagingService.publish('editor.dirtyStateChanged', { path: 'yet/another', dirty: true });
+      messagingService.publish('editor.dirtyStateChanged', { path: 'and/yet/another', dirty: true });
+      messagingService.publish('editor.dirtyStateChanged', { path: 'and/still/another', dirty: true });
+      messagingService.publish('editor.close', { id: 'some/file' }); // not open not dirty
```

Comment:
You could use the file paths to express their intended state / use in the test scenario. You'd get rid of the comments, and it would be easier to keep track, and cross-check with the expected result in the _then_ block.